lib/src/modelinfo/modelentity.dart

```diff
@@ -7,63 +7,85 @@ import 'modelproperty.dart';
 /// ModelEntity describes an entity of a model and consists of instances of `ModelProperty` as well as an other entity
 /// information: id, name and last property id.
 class ModelEntity {
-  IdUid id, lastPropertyId;
-  String name;
-  int flags;
-  List<ModelProperty> properties;
-  ModelProperty idProperty;
-  ModelInfo _model;
+  IdUid id;
+  IdUid lastPropertyId = IdUid.empty();
+  /*late*/ String _name;
+  int _flags = 0;
+  final _properties = <ModelProperty>[];
+  ModelProperty /*?*/ _idProperty;
+  final ModelInfo /*?*/ _model;
+
+  String get name => _name;
+
+  set name(String /*?*/ value) {
+    if (value == null || value.isEmpty) {
+      throw Exception('name must not be null or an empty string');
+    }
+    _name = value /*!*/;
```

Comment:
I agree it is odd because of the check just above... However, it was suggested like this by the migration tool :shrug: Hopefully they improve it later on and we can get rid of those.

Reply:
I agree it is odd because of the check just above... However, it was suggested like this by the migration tool shrug Hopefully they improve it later on and we can get rid of those.